### ISSUE TITLE
launcher binary could be 4x smaller

### ISSUE THREAD

**Issue by dgageot** [buildpacks/lifecycle#247]
I was surprised that the `launcher` binary is 8.7Mb. So I extracted just the code it uses, into a new project: https://github.com/dgageot/smaller-lifecycle and now, the binary is just 2Mb. 🎆

**Comment by dgageot** [comment]
Running the `launcher` executes 17 `init()` functions from imported packages

**Comment by sclevine** [comment]
Strongly agree that we should fix this. @hone was working on a rust implementation, but for now, maybe we could factor the launcher (and significant dependent code) into modules that build a smaller binary?

**Comment by dgageot** [comment]
@sclevine Not sure what you mean by "factor the launcher into modules". I tried to cut this project in multiple packages but gave up. Maybe I could try again.

**Comment by sclevine** [comment]
Factoring the launcher out into a separate module that doesn't depend on lifecycle would also allow folks to build it without pulling all of github.com/docker/docker. Thinking about that more, it doesn't matter as much now that the exporter adds the launcher to the run image at build time.

Here's an initial pass at factoring out packages to minimize launcher dependencies: https://github.com/buildpacks/lifecycle/pull/255

```
 opal:lifecycle stephen$ du -h out/lifecycle/launcher
2.2M	out/lifecycle/launcher
```

**Comment by ekcasey** [comment]
resolved by #255
